### PR TITLE
perf: enhance account key update logic to batch transactions instead of loading all with sleep

### LINKED ISSUE #59: Incorrect status reported by account manager
Although accounts couldn't be updated, it reports as successful. This happens if we rerun `solo node start` second time:

Screenshot 2024-02-23 at 12 01 56 AM

### LINKED ISSUE #71: Refactor update special account keys using Listr
Currently it uses an array of promises to attempt to update all account keys at the same time. It would have been better to use Listr and create separate sub-tasks where we can control sequential or parallel execution as well as provide better updates in the console for user.

### LINKED ISSUE #76: Remove jest/test-sequencer from package.json as it is no longer used

### LINKED ISSUE #104: Enhance account key update logic to batch transactions instead of loading all with sleep

### PR DESCRIPTION
**Pull request by jeromy-cannon**
## Description

This pull request changes the following:

* enhance account key update logic to batch transactions instead of loading all with sleep
* removed unused reference to `@jest/test-sequencer`
* added logic to properly close WebSocketServer connections
* removed multiple sleep commands
* updated for check for `Terminating Netty` string
* removed extra creation of NodeClient to simplify and increase performance
* restructured mirror node call to better leverage listr2 subtasks for user experience
* restructured update of account keys to better leverage listr2 subtasks
* fixed bug was not reporting skips and rejections correctly in the account key updates loop

### Related Issues

* Closes #104 
* Closes #76
* Closes #71 
* Closes #59

### CODE REVIEW

File: src/commands/node.mjs [hunk 1]
```diff
@@ -502,53 +506,125 @@ export class NodeCommand extends BaseCommand {
...
+          if (ctx.config.deployMirrorNode) {
+            const subTasks = [
+              {
+                title: 'Wait for proxies to verify node servers are running',
+                task: async (ctx, _) => {
+                  await sleep(15000) // give time for haproxy to detect that the node server is up on grpc port, queries every 10 seconds
+                }
+              },
+              {
+                title: 'Get the mirror node importer address book',
```
Comment (leninmehedy) [inline src/commands/node.mjs:518]: `Get the mirror node importer address book` -> `Prepare address book`

File: src/commands/node.mjs [hunk 2]
```diff
@@ -502,53 +506,125 @@ export class NodeCommand extends BaseCommand {
...
+                task: async (ctx, _) => {
+                  ctx.addressBook = await self.getAddressBook(ctx.nodeClient)
+                  ctx.config.valuesArg += ` --set "hedera-mirror-node.importer.addressBook=${ctx.addressBook}"`
+                }
+              },
+              {
+                title: `Upgrade chart '${constants.FULLSTACK_DEPLOYMENT_CHART}'`,
```
Comment (leninmehedy) [inline src/commands/node.mjs:525]: `Upgrade chart ..` ->`Deploy mirror node`

File: src/commands/node.mjs [hunk 3]
```diff
@@ -502,53 +506,125 @@ export class NodeCommand extends BaseCommand {
...
+                  await self.chartManager.upgrade(
+                    ctx.config.namespace,
+                    constants.FULLSTACK_DEPLOYMENT_CHART,
+                    ctx.config.chartPath,
+                    ctx.config.valuesArg
+                  )
+                }
+              },
+              {
+                title: 'Waiting for explorer pod to be ready',
```
Comment (leninmehedy) [inline src/commands/node.mjs:536]: `Waiting for explorer pod to be ready` -> `Waiting for Hedera Explorer to be ready`

File: src/core/account_manager.mjs [hunk 4]
```diff
@@ -115,36 +118,54 @@ export class AccountManager {
...
-    await this.stopPortForwards()
+    return batchSets
   }
 
   /**
    * stops and closes all of the port forwards that are running
    * @returns {Promise<void>}
    */
   async stopPortForwards () {
+    global.accountManagerPortForwardClosedCount = 0 // global variable to be used for WebSocketServer callback
```
Comment (leninmehedy) [inline src/core/account_manager.mjs:149]: We should avoid global namespace, instead we could create a static class variable (`AccountManager.portForwardClosed`) to track it.

**Review by leninmehedy — CHANGES_REQUESTED** [review]

**Review by leninmehedy — APPROVED** [review]
Please address my comments and merge when you are ready.

File: src/commands/node.mjs (hunk 1 above)
Comment (jeromy-cannon) [reply src/commands/node.mjs]: done

File: src/commands/node.mjs (hunk 2 above)
Comment (jeromy-cannon) [reply src/commands/node.mjs]: done

File: src/commands/node.mjs (hunk 3 above)
Comment (jeromy-cannon) [reply src/commands/node.mjs]: done

File: src/core/account_manager.mjs (hunk 4 above)
Comment (jeromy-cannon) [reply src/core/account_manager.mjs]: done

File: src/commands/node.mjs [hunk 5]
```diff
@@ -502,50 +502,132 @@ export class NodeCommand extends BaseCommand {
...
-              title: 'Waiting for explorer pod to be ready',
-              task: async (ctx, _) => {
-                if (ctx.config.deployHederaExplorer) {
-                  await this.k8.waitForPod(constants.POD_STATUS_RUNNING, [
-                    'app.kubernetes.io/component=hedera-explorer', 'app.kubernetes.io/name=hedera-explorer'
-                  ], 1, 100)
+          if (ctx.config.deployMirrorNode) {
+            const subTasks = [
+              {
+                title: 'Wait for proxies to verify node servers are running',
```
Comment (leninmehedy) [inline src/commands/node.mjs:508]: Status message could be simplified: 
```
✔ Check node proxies are ACTIVE
  ✔ Check node proxy: node0
  ✔ Check node proxy: node1
  ✔ Check node proxy: node2
  ```

File: src/core/k8.mjs [hunk 6]
```diff
@@ -33,6 +33,8 @@ import { V1ObjectMeta, V1Secret } from '@kubernetes/client-node'
  * For parallel execution, create separate instances by invoking clone()
  */
 export class K8 {
+  static _webSocketGetters = []
```
Comment (leninmehedy) [inline src/core/k8.mjs:36]: Would be nice to have some description of the variable. It was not clear what it is.

File: src/commands/node.mjs [hunk 7]
```diff
@@ -557,36 +639,57 @@ export class NodeCommand extends BaseCommand {
...
+        if (logResponse.body.includes('Server be_servers/server1 is UP')) {
+          return true
+        }
+
+        attempts++
+        this.logger.debug(`Checking for pod ${podName} to realize network node is UP [attempt: ${attempts}/${maxAttempts}]`)
+        await sleep(1000)
+      }
+    } else {
+      throw new FullstackTestingError('TBD')
```
Comment (leninmehedy) [inline src/commands/node.mjs:676]: TBD -> `proxy for 'node0' is not ACTIVE [ attempt = 50/50 ]`

File: src/commands/node.mjs (hunk 5 above)
Comment (jeromy-cannon) [reply src/commands/node.mjs:508]: done

File: src/core/k8.mjs (hunk 6 above)
Comment (jeromy-cannon) [reply src/core/k8.mjs:36]: done

File: src/commands/node.mjs (hunk 7 above)
Comment (jeromy-cannon) [reply src/commands/node.mjs:676]: done

**Review by leninmehedy — APPROVED** [review]

**Pull request merged**